adminpages/pagesettings.php

```diff
@@ -21,7 +21,32 @@
  * @since 1.8.5
  */
 $extra_pages = apply_filters('pmpro_extra_page_settings', array());
-$post_types = apply_filters('pmpro_admin_pagesetting_post_type_array', array( 'page' ) );
+
+/**
+ * @deprecated replaced with pmpro_admin_pagesetting_post_type from 2.6.6
+ */
+$post_types = apply_filters( 'pmpro_admin_pagesetting_post_type_array', array( 'page' ) );
+
+// For backward compatibility we extract the first element from the array
+$post_types = is_array( $post_types ) ? $post_types[ array_key_first( $post_types ) ] : $post_types;
+
+// Set default post type for page settings dropdown.
+$post_type_default = 'page';
+
+// Get post type from historic hook and reset to default if it does not exists and is not hierarchical
+$post_type = post_type_exists( $post_types ) && is_post_type_hierarchical( $post_types ) ? $post_types : $post_type_default;

```

Comment:
This isn't necessary because wp_dropdown_pages will do this logic and fallback to page post type